File: projekty/PO_Projekt3/antylopa.py
from zwierze import Zwierze

class Antylopa(Zwierze):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 4, 4, 1.00, 2, 0.50, 1.00)

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Antylopa)

    def stworzDziecko(self, cel):
        return Antylopa(self._swiat, cel)

    def rysowanie(self):
        return (217, 186, 140)

File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Zwierzeta/Antylopa.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Zwierzeta;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Antylopa extends Zwierze {

	public Antylopa(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Antylopa.class), Stale.inicjatywa.get(Antylopa.class),
				Stale.szansaRuchu.get(Antylopa.class), Stale.zasiegRuchu.get(Antylopa.class),
				Stale.szansaUcieczki.get(Antylopa.class), Stale.szansaRozmnazania.get(Antylopa.class));
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Antylopa;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Antylopa(swiat, cel);
	}

}
